### DISCUSSION TITLE
Course building ideas

### DISCUSSION THREAD

**Discussion by Helezak** [Courseplay/Courseplay_FS22#3186] [Ideas]
I would like to make a request. I hope the developer of Courseplay reads this.
I do a lot of funky shaped fields and as good as the course builder is, sometimes my field is just to messy and I need to play with it.
I have one field in mind that has bulges with narrow places between them. The up and down in the bulges is great but the narrows often compress the up and down to the extent that they are not navigable. Based on the shape some parts need an east west orientation while other parts really need a north south.
There is a custom field where you can plot out the field and their is a way to drive a custom field, but the only way to build a course is to use the course editor which is great, but cumbersome.  Hand creating a course in vineyards is crazy hard.  I use it a lot, but it would be nice to cut and paste pieces of a course. Or to drive a course and have it recorded. I have done that with Autodrive but with different width implements on the same field the map gets very messy. I would also like to link courses so after one ends they jump to another one. This can be done if I could cut and paste route sections.

If you could record a path, I could use Autodrive and Courseplay together since their course development tools do complement each other. I could let courseplay create headlands and up and downs one way and then change directions and do it again and record them in autodrive and use its editor to connect sections.

Anyway, not complaining just wishing.

**Comment by Tensuko** [comment]
Sry but recording a fieldwork course won't happen as it leads to user made fiels wich then can cause problems.
Instead of asking for solutions, you could actually show us your fields.
If they are too complicated, there is now way that we will make it possible.
But without see what they look like, no Idea what we COULD do.

> I would also like to link courses so after one ends they jump to another one.

Won't work as well, our course have many informations in it that the user doesn't see and don't need but we do need them for the tools.
Thats also another reason why recording won't happen.

**Comment by Helezak** [comment]
![mars](https://github.com/Courseplay/Courseplay_FS22/assets/165971535/be7705a1-5380-4f5f-82cd-24f83f18b11f)
This is the  shape file on the Mars map I have the most trouble with.  I can currently have the course made which ends up with collisions with the walls and the domes so i use the editor to alter the saved course.  But If I could just drive my tractor and have the path recorded that would work.

**Reply by Tensuko** [reply]
>  But If I could just drive my tractor and have the path recorded that would work.

Again, that won't happen.

What work width do you use there?
I am pretty sure with 4m and not too long tools (the best is 3point attached for this) it could easy work.
If you expect high work width with big tools, then I can tell you it will never work.

You can also try differen field patterns like spiral.

**Comment by Helezak** [comment]
I am just doing grass here so most implements are small except the windrow and fertilizer spreader.
For a 7m implement, set on automatic, produces the following for 0 3 and 5 headlands.  The zero headlands does not work as the tractor runs into the obstacles on the sides.  Just about every other pattern I choose has transition tracks that pass through the obstacles and need to be edited.

**Comment by Helezak** [comment]
![mars 4 view](https://github.com/Courseplay/Courseplay_FS22/assets/165971535/d04b12e7-51d8-462d-8f4f-bcc74d059374)

**Reply by Tensuko** [reply]
the 2nd option looks totaly fine
would take that
tuning can be done with the editor, no other option

**Comment by Helezak** [comment]
Again, the tool does a nice job but it can use some tweeking and sometimes just driving the course would take less time.
